pkg/operator/sync.go

```diff
@@ -244,6 +246,93 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 		return fmt.Errorf("refusing to read images.json version %q, operator version %q", imgs.ReleaseVersion, optrVersion)
 	}
 
+	// handle image registry certificates.
+	// parse these, add them to ctrlcfgspec and then handle these in the daemon write to disk function
+	cfg, err := optr.imgLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+	imgRegistryUsrData := []v1.ImageRegistryBundle{}
+	if cfg.Spec.AdditionalTrustedCA.Name != "" {
+		cm, err := optr.clusterCmLister.ConfigMaps("openshift-config").Get(cfg.Spec.AdditionalTrustedCA.Name)
+		if err != nil {
+			klog.Warningf("could not find configmap specified in image.config.openshift.io/cluster with the name %s", cfg.Spec.AdditionalTrustedCA.Name)
+		} else {
+			for key, d := range cm.Data {
+				raw, err := base64.StdEncoding.DecodeString(d)
+				if err != nil {
+					imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+						File: key,
+						Data: []byte(d),
+					})
+				} else {
+					imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+						File: key,
+						Data: raw,
+					})
+				}
+			}
+			for key, bd := range cm.BinaryData {
+				imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+					File: key,
+					Data: bd,
+				})
+			}
+		}
+	}
+
+	imgRegistryData := []v1.ImageRegistryBundle{}
+	cm, err := optr.clusterCmLister.ConfigMaps("openshift-config-managed").Get("image-registry-ca")
+	if err != nil {
+		klog.Info("the image registry is not installed so the MCO will not distribute its certificates.")
```

Comment:
This message is probably unnecessary every sync (which is how often it comes out now). 

I also was using a cluster that does have the registry installed, and Managed even and it still tells me: 

```
I0722 02:02:25.905658       1 sync.go:287] the image registry is not installed so the MCO will not distribute its certificates.
W0722 02:02:28.340362       1 warnings.go:70] unknown field "spec.dns.spec.platform"
I0722 02:02:31.370855       1 status.go:376] Condition: Updated, Machines: 3, UpdatedMachines: 3, DegradedMachines: 0, UnavailableMachines: 0
I0722 02:02:31.370879       1 status.go:376] Condition: Updated, Machines: 3, UpdatedMachines: 3, DegradedMachines: 0, UnavailableMachines: 0
I0722 02:02:31.574548       1 status.go:376] Condition: Updated, Machines: 3, UpdatedMachines: 3, DegradedMachines: 0, UnavailableMachines: 0
I0722 02:02:31.574568       1 status.go:376] Condition: Updated, Machines: 3, UpdatedMachines: 3, DegradedMachines: 0, UnavailableMachines: 0
I0722 02:02:31.992175       1 sync.go:287] the image registry is not installed so the MCO will not distribute its certificates.
W0722 02:02:34.426566       1 warnings.go:70] unknown field "spec.dns.spec.platform"
```
So if you do want to keep it, I might make it more "truthy" ?  (This might come out in the wash with whatever else you're going to work out with the image registry team, but I'll mention it anyway just in case)

Reply:
We can get rid of this, I guess it is a no-op either way. The reason you may have gotten that message still is because you may be running an openshift version where the CIRO does not create the configmap, since that just merged